script.js

```diff
@@ -1,12 +1,244 @@
+  const term = event.target.value.toLowerCase();
+  
+    let filtered = allEpisodes.filter(episode =>{
+
+        return (episode.name + episode.summary).toLowerCase().includes(term);
+      
+      })
+      
+      displayParagraph.innerText = `Displaying:${filtered.length}/${allEpisodes.length}`;
+      
+      filtered.forEach(episode =>{
+
+        createEpisodeCards(episode);
+        console.log(episode.name + "-->"+episode.summary);
+  
+      })
+  
+      
+
+}
+
+        /*function to create the dropdown list for shows */
+
+function createShowList(showsList){
+
+  showsList.forEach(show =>{
+    const optEl = document.createElement("option");
+    optEl.value = show.id;
+    optEl.textContent = show.name;
+    showsDropDownEl.appendChild(optEl);
+
+  })
+}        
+
+          /*function to select a show */
+function selectShow(){
+  let SHOW_ID = event.target.value;
+  const episodeSelectEl = document.querySelector("#all-episodes");
+  console.log(episodeSelectEl);
+  
+  fetch(`https://api.tvmaze.com/shows/${SHOW_ID}/episodes`)
+  .then(response => response.json())
+  .then(data => {
+    allEpisodes = data;  
+    //console.log(allEpisodes.names);
+    makePageForEpisodes(allEpisodes);
+    searchBar.addEventListener('keyup',doLiveSearch);
+    createDropDownList(allEpisodes);
+    dropDownEl.addEventListener('change',selectEpisode);
+  })
+  .catch(err => console.log(err));
+
+} 
+
+           /*function to create the dropdown list for episodes */
+function createDropDownList(episodeList){
+  let prevEpisodes = document.querySelectorAll("#episode-select option");
+  for(let i = 1; i < prevEpisodes.length; i++){
+    dropDownEl.removeChild(prevEpisodes[i]);
+  }
+  console.log(prevEpisodes);
+
+  episodeList.forEach(episode => {
+    let optionEl = document.createElement("option");
+    optionEl.value = episode.name;
+    let seasonNumber = episode.season;
+    let episodeNumber = episode.number;
+    let sE = addZeros(seasonNumber,episodeNumber);
+    optionEl.textContent = " ";
+    optionEl.textContent = sE + " - " + episode.name;
+    //console.log(optionEl);
+    dropDownEl.appendChild(optionEl);
+    //console.log(sE + " - " + episode.name);
+
+  })
+ 
+
+}  
+
+         
+
+            /*function to select an episode */
+function selectEpisode(){
+
+  const selected = event.target.value;
+  const episodeCards = document.querySelectorAll(".episodes-card");
+  const episodeTitles = document.querySelectorAll("h3");
+  
+  for(let i = 0; i < episodeCards.length; i++){
+    if(selected != "showAll"){
+
+      if(episodeTitles[i].innerHTML.includes(selected)){
+        //console.log(episodeCards[i]);
+        episodeCards[i].style.display = 'block';
+      }else{
+        episodeCards[i].style.display = 'none';
+      }
+    }else{
+      episodeCards[i].style.display = 'inline';
+    }
+  }
+  
+}            
+
+
+
+                /*function to create episode cards*/
+
+function createEpisodeCards(episode){
```

Comment:
```suggestion
function createEpisodeCard(episode){
```
Keep this function name singular - it is a good function as it keeps the logic for a building an episode in a separate place